CMakeLists.txt

```diff
@@ -290,6 +290,13 @@ IF(BUILD_LLVM_DWARF)
   SET(binaryen_objs ${binaryen_objs} $<TARGET_OBJECTS:llvm_dwarf>)
 ENDIF()
 
+# Code generation.
+
+add_custom_target(gen_exprs ALL
```

Comment:
it seems like you should just be able to use `add_custom_command` (using `OUTPUT` instead of `BYPRODUCTS`, and gen-exprs.py as the `MAIN_DEPENDENCY`), and then it wouldn't have to run everytime. Also the targets that depend on its output could just depend on whatever file they use, instead of this target. And you could remove the special no-change logic from the script itself.

Reply:
I tried that path earlier, but failed. The problem I ran into is that this tool emits `.h` files, and CMake doesn't track dependencies on them the way it does for other things. (Maybe related to that in CMake a C++ file doesn't write out which headers it depends on?)

I'm not an expert on how CMake works, but I spent a bunch of time on this and did quite a bit of searching for an example of how to do it (there are plenty of examples for generating a **source** file, but not a header), but came up empty.. :disappointed: